### PR TITLE
[framework] implementations of GridFactoryInterface are more modular now

### PR DESCRIPTION
**Pull request by MattCzerner**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Slight change of behavior of GridFactory required to fork whole class, this PR modularize GridFactory so you only change/extend the part that you are interested in changing.
|New feature| No
|[BC breaks](/docs/contributing/backward-compatibility-promise.md)| Yes
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/src/Model/Product/Flag/FlagGridFactory.php [hunk 1]
```diff
@@ -45,15 +47,32 @@ public function __construct(
...
+    protected function getDataSource(): DataSourceInterface
     {
         $queryBuilder = $this->em->createQueryBuilder();
         $queryBuilder
             ->select('a, at')
             ->from(Flag::class, 'a')
             ->join('a.translations', 'at', Join::WITH, 'at.locale = :locale')
             ->setParameter('locale', $this->localization->getAdminLocale());
-        $dataSource = new QueryBuilderDataSource($queryBuilder, 'a.id');
+        return new QueryBuilderDataSource($queryBuilder, 'a.id');
```
Comment (vitek-rostislav) [inline packages/framework/src/Model/Product/Flag/FlagGridFactory.php]:
```suggestion

        return new QueryBuilderDataSource($queryBuilder, 'a.id');
```

**Comment by vitek-rostislav** [comment]
Hi @MattCzerner, after some testing and prototyping, it seems to me, that it would be sufficient to extract just the data source creation to a protected method, see https://github.com/shopsys/shopsys/commit/d7e4b172dd85f5b64c0f84e3a12cea0ad01bb88c.

I would like to discuss this, so lets call each other :phone: :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
We decided not to continue with this PR now. First, we will document the Grid component in general, investigate use cases how the grids are modified on real projects a then return to the refactoring which will be based on the collected information.

**Comment by boris-brtan** [comment]
hi @vitek-rostislav, is this branch still needed.

**Pull request closed without merging**